app/models/checkout.rb

```diff
@@ -0,0 +1,21 @@
+class Checkout
+
+  attr_reader :cart
+
+  def initialize(cart:, pricing_rules: Discount)
+    @cart = cart
+    @pricing_rules = pricing_rules
+  end
+
+  def add_item(item)
+    cart.add_item(item, pricing_rules)
+  end
+
+  def total
+    cart.line_items.sum(:total).to_f
+  end
+
+  private
+
+  attr_reader :pricing_rules
+end
```

Comment:
This class sits at the top most level and is responsible for executing the end user's request related to the checkout process. In theory this class would be expanded to encompass more functionality in the future as this is a composition of several components working together to full the process of a **checkout**.

Because this is a composition class, there is no need to make it an AR model.